src/core/exec-invoke.c

```diff
@@ -2280,6 +2280,56 @@ static int create_many_symlinks(const char *root, const char *source, char **sym
         return 0;
 }
 
+static int is_idmapping_supported(const char *path) {
+        _cleanup_close_ int mount_fd = -EBADF, userns_fd = -EBADF, dir_fd = -EBADF;
+        _cleanup_free_ char *uid_map = NULL, *gid_map = NULL;
+        int r;
+
+        assert(path);
+
```

Comment:
can you add a quick:

```c
if (!mount_new_api_supported())
        return false;
```

here please. that test is cached after all and might save us a lot of work.